### PR TITLE
Updated System Stack Create tests 

### PR DESCRIPTION
**Pull request by fuddstar**
Refactored system stack test to allow for different yaml files to be passed in to allow for different tests.

Used code from amz.py

Updated changelog.md

BUILD Passing https://gaautobots.atlassian.net/builds/browse/AM-AST11-8

### CODE REVIEW

File: test/sys_tests/test_sys_stack_create.py [hunk 1]
```diff
@@ -1,108 +1,181 @@
...
+            print('Error occurred creating AWS CloudFormation stack and returned status code {0}.'.format(stack_status))
+            exit(1)
+        else:
+            print('Stack Status: {0}'.format(stack_status))
+        time.sleep(time_delay)
+
+
+def main():
+    """
+
```
Comment (snayler0) [inline test/sys_tests/test_sys_stack_create.py]: Docstrings?

File: test/sys_tests/test_sys_stack_create.py [hunk 2]
```diff
@@ -1,108 +1,181 @@
...
+    :return:
+    """
+    s3_client.upload_file(template_path, s3_bucket, s3_key)
+
+    print('File Successfully Uploaded to S3')
+
+
+def create_and_delete_stack(cf_client, stack_name, s3_bucket, s3_key):
+    """
+    his Script will take a cloud formation template file and upload it to create a cloud formation stack in aws using boto3
```
Comment (snayler0) [inline test/sys_tests/test_sys_stack_create.py]: This*....... :D

File: test/sys_tests/test_sys_stack_create.py [hunk 3]
```diff
@@ -1,108 +1,181 @@
...
+    http://boto3.readthedocs.org/en/latest/reference/services/cloudformation.html#CloudFormation.Client.create_stack
+    :param cf_client:
```
Comment (snayler0) [inline test/sys_tests/test_sys_stack_create.py]: params definitions?

File: test/sys_tests/test_sys_stack_create.py [hunk 4]
```diff
@@ -1,108 +1,181 @@
...
-    elif stack_status in ('ROLLBACK_IN_PROGRESS', 'ROLLBACK_COMPLETE'):
-        print('Error occurred creating AWS CloudFormation stack and returned status code {0}.'.format(stack_status))
-        exit(1)
-    else:
-        print('Stack Status: {0}'.format(stack_status))
-    time.sleep(time_delay)
+def read_yaml(user_yaml):
+    """
+    Ingest user YAML
+    :param user_yaml:
```
Comment (snayler0) [inline test/sys_tests/test_sys_stack_create.py]: params definitions? (I know it's pretty straight forward but something something opensource something.... :) )

File: test/sys_tests/test_sys_stack_create.py [hunk 5]
```diff
@@ -1,108 +1,181 @@
...
+
+    with open(template_path, 'w') as template_file:
+        template_file.write(template_data)
+        template_file.close()
+
+
+def upload_s3(s3_client, template_path, s3_bucket, s3_key):
+    """
+    Upload to s3
+    :param s3_client:
```
Comment (snayler0) [inline test/sys_tests/test_sys_stack_create.py]: params definitions?

File: test/sys_tests/test_sys_stack_create.py (hunk 1 above)
Comment (fuddstar) [inline test/sys_tests/test_sys_stack_create.py]: yep good point :) I'll add some

**Comment by GypsyBojangles** [comment]
Could we get a nicer PR name? :)

**Comment by GypsyBojangles** [comment]
Approved

**Comment by snayler0** [comment]
Approved

**Pull request merged**